source/extensions/filters/http/ip_tagging/ip_tagging_filter.cc

```diff
@@ -50,16 +122,76 @@ IpTaggingFilterConfig::IpTaggingFilterConfig(
     }
 
     tag_data.emplace_back(ip_tag.ip_tag_name(), cidr_set);
-    stat_name_set_->rememberBuiltin(absl::StrCat(ip_tag.ip_tag_name(), ".hit"));
   }
-  trie_ = std::make_unique<Network::LcTrie::LcTrie<std::string>>(tag_data);
+  return tag_data;
 }
 
-void IpTaggingFilterConfig::incCounter(Stats::StatName name) {
-  Stats::SymbolTable::StoragePtr storage = scope_.symbolTable().join({stats_prefix_, name});
-  scope_.counterFromStatName(Stats::StatName(storage.get())).inc();
+std::shared_ptr<TagSetWatcher>
+TagSetWatcher::create(Server::Configuration::FactoryContext& factory_context, std::string filename,
+                      Stats::Scope& scope, const std::string& stat_prefix) {
+
+  auto ptr =
+      std::make_shared<TagSetWatcher>(factory_context, std::move(filename), scope, stat_prefix);
+  return ptr;
+}
+
+TagSetWatcher::TagSetWatcher(Server::Configuration::FactoryContext& factory_context,
+                             Event::Dispatcher& dispatcher, Api::Api& api, std::string filename,
+                             Stats::Scope& scope, const std::string& stat_prefix)
+    : api_(api), filename_(filename), watcher_(dispatcher.createFilesystemWatcher()), scope_(scope),
+      stat_prefix_(stat_prefix), factory_context_(factory_context),
+      yaml_(absl::EndsWith(filename, MessageUtil::FileExtensions::get().Yaml)) {
+
+  const auto split_path = api_.fileSystem().splitPathFromFilename(filename);
+  watcher_->addWatch(absl::StrCat(split_path.directory_, "/"), Filesystem::Watcher::Events::MovedTo,
+                     [this]([[maybe_unused]] std::uint32_t event) { maybeUpdate_(); });
+
+  maybeUpdate_(true);
 }
 
+// read the file from filesystem, load the content and only update if the hash has changed.
+void TagSetWatcher::maybeUpdate_(bool force) {
+  std::string contents = api_.fileSystem().fileReadToEnd(filename_);
+
+  uint64_t hash = 0;
+  hash = HashUtil::xxHash64(contents, hash);
```

Comment:
nit: `uint64_t hash = HashUtil::xxHash64(contents, hash);` ?

Reply:
So I looked into this, and it turns out this results into a compiler error with `variable 'hash' is uninitialized when used within its own initialization`.

Reply:
Oh right -- sorry didn't see it was passed as a param. Thanks for checking.